packages/hash/frontend/theme-override.d.ts

```diff
@@ -1,237 +1 @@
-    regularTextParagraphs?: React.CSSProperties;
-    regularTextLabels?: React.CSSProperties;
-    smallTextParagraphs?: React.CSSProperties;
-    smallTextLabels?: React.CSSProperties;
-    microText?: React.CSSProperties;
-  }
-}
-
-declare module "@mui/material/Typography" {
-  interface TypographyPropsVariantOverrides {
-    title: true;
-    h1: true;
-    h2: true;
-    h3: true;
-    h4: true;
-    h5: true;
-    mediumCaps: true;
-    smallCaps: true;
-    largeTextLabels: true;
-    regularTextPages: true;
-    regularTextParagaphs: true;
-    regularTextLabels: true;
-    smallTextParagraphs: true;
-    smallTextLabels: true;
-    microText: true;
-    // disable unused defaults
-    h6: false;
-    subtitle1: false;
-    subtitle2: false;
-    body1: false;
-    body2: false;
-    caption: false;
-    button: false;
-    overline: false;
-  }
-}
-
-declare module "@mui/material/Button" {
-  interface ButtonPropsVariantOverrides {
-    primary: true;
-    secondary: true;
-    tertiary: true;
-    tertiary_quiet: true;
-    warning: true;
-    danger: true;
-    // Disable defaults
-    contained: false;
-    outlined: false;
-    text: false;
-  }
-
-  interface ButtonPropsColorOverrides {
-    purple: true;
-    teal: true;
-    warning: true;
-    danger: true;
-    gray: true;
-    // Disable defaults
-    primary: false;
-    secondary: false;
-    success: false;
-    error: false;
-    info: false;
-    warning: false;
-  }
-
-  interface ButtonPropsSizeOverrides {
-    large: true;
-    medium: true;
-    small: true;
-    xs: true;
-  }
-}
-
-declare module "@mui/material/IconButton" {
-  interface IconButtonPropsSizeOverrides {
-    large: true;
-    medium: true;
-    small: true;
-    xs: true;
-  }
-}
-
-declare module "@mui/lab/TreeItem" {
-  interface TreeItemContentProps {
-    expandable: boolean;
-    url: string;
-    depth: number;
-  }
-}
-
-declare module "@mui/material/TextField" {
-  interface TextFieldPropsSizeOverrides {
-    xs: true;
-    small: true;
-    medium: true;
-    large: true;
-  }
-}
-
-declare module "@mui/material/InputBase" {
-  interface InputBasePropsSizeOverrides {
-    xs: true;
-    small: true;
-    medium: true;
-    large: true;
-  }
-}
-
-declare module "@mui/material/Chip" {
-  interface ChipPropsSizeOverrides {
-    xs: true;
-    small: true;
-    medium: true;
-    large: true;
-  }
-
-  interface ChipPropsColorOverrides {
-    gray: true;
-    blue: true;
-    purple: true;
-    red: true;
-    orange: true;
-    green: true;
-    yellow: true;
-    pink: true;
-    teal: true;
-    mint: true;
-    navy: true;
-    // Disable defaults
-    default: false;
-    primary: false;
-    secondary: false;
-    success: false;
-    error: false;
-    info: false;
-    warning: false;
-  }
-}
-
-// eslint-disable-next-line import/no-default-export -- @see https://github.com/mui-org/material-ui/issues/28244
-export default "";
+import "@hashintel/hash-design-system/theme-override.d";
```

Comment:
It might be possible to reduce PR diff if we rename this file. Then most red lines will count as renaming and `packages/hash/design-system/src/theme-override.d.ts` won't be treated as the new file. Perhaps, there's another `d.ts` file we can add the import to?

The same trick with git diff (git blame) applies to `packages/hash/frontend/src/shared/ui/(button|menu-item).tsx`. Perhaps, we give `shared/ui` some temp name (like `app-ui` or `ui-overrides`), save git blame and rename those files back back in the future. WDYT?

Reply:
Thanks! Just tried your first suggestion and it reduced the diff massively!